### ISSUE TITLE
Trying to output as .vhd fails

### ISSUE THREAD

**Issue by radu-matei** [linuxkit/linuxkit#1836]
**Description**

Docker info
Client:
 Version:      17.05.0-ce
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 21:43:09 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 21:43:09 2017
 OS/Arch:      linux/amd64
 Experimental: true

Moby info
moby version 0.0
commit: 318cf6841cba114ea2fc829dfb9ebd6f2e24632c

LinuxKit info
linuxkit version 0.0
commit: 318cf6841cba114ea2fc829dfb9ebd6f2e24632c

When adding the output format to .vhd, `moby build test.yml` remains hanging.

**Steps to reproduce the issue:**
Basically start with the basic sshd example, then add .vhd as output.

**Describe the results you received:**
Everything hangs no matter how much I allow it to finish

**Describe the results you expected:**
I would like to have a .vhd as output

**Comment by rneugeba** [comment]
creating a VHD requires a lot of memory and it takes some time, but works on my machine(s):
```
% time moby build linuxkit.yml
Extract kernel image: linuxkit/kernel:4.9.x
Add init containers:
Process init image: linuxkit/init:b3740303f3d1e5689a84c87b7dfb48fd2a40a192
Process init image: linuxkit/runc:47b1c38d63468c0f3078f8b1b055d07965a1895d
Process init image: linuxkit/containerd:cf2614f5a96c569a0bd4bd54e054a65ba17d167f
Process init image: linuxkit/ca-certificates:3344cdca1bc59fdfa17bd7f0fcbf491b9dbaa288
Add onboot containers:
  Create OCI config for linuxkit/sysctl:2cf2f9d5b4d314ba1bfc22b2fe931924af666d8c
  Create OCI config for linuxkit/binfmt:131026c0cf6084467316395fed3b358f64bda00c
  Create OCI config for linuxkit/dhcpcd:2def74ab3f9233b4c09ebb196ba47c27c08b0ed8
Add service containers:
  Create OCI config for linuxkit/rngd:61a07ced77a9747708223ca16a4aec621eacf518
  Create OCI config for nginx:alpine
Add files:
  etc/docker/daemon.json
Create outputs:
  linuxkit-kernel linuxkit-initrd.img linuxkit-cmdline
  linuxkit.vhd

real    2m16.034s
user    0m9.805s
sys     0m4.014s
```

**Comment by radu-matei** [comment]
@rneugeba - Ok, in my case I left it for a couple of minutes, but didn't finish. If I then convert the image using `qemu-img`, then it is rather fast. I will surely leave it more time, but do you know why it's taking so much longer than compared to an .iso, for example?

Thanks!

**Comment by justincormack** [comment]
Can you try running the parts of the vhd script which is https://github.com/linuxkit/linuxkit/blob/master/tools/mkimage-vhd/make-vhd by hand (not in a container and maybe inside a container too) and seeing what part is slow.

**Comment by radu-matei** [comment]
Update: I rebuilt the tool and created again the image with VHD output and now it works. 
The issue now is that the output VHD's size is huge compared to the ISO (1 GB vs 115 MB).

**Comment by justincormack** [comment]
I have been modifying the code to remove the hard coded 1GB default in https://github.com/moby/tool/pull/49 but still improving it.
